### ISSUE TITLE
'warnings by module' table missing in a report

### ISSUE THREAD

**Issue by zasca** [pylint-dev/pylint#9145]
### Bug description

```python
Table 'warnings by module' missing in a report
```

### Command used

```shell
echo foo > module1.py
echo bar > module1.py
pylint -ry module1.py module2.py
```

### Pylint output

```python
************* Module module1
module1.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module1.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module1.py:1:0: E0602: Undefined variable 'foo' (undefined-variable)
************* Module module2
module2.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module2.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module2.py:1:0: E0602: Undefined variable 'bar' (undefined-variable)


Report
======
2 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |2      |NC         |NC         |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|method   |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|function |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+



6 lines have been analyzed

Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |4      |66.67 |NC       |NC         |
+----------+-------+------+---------+-----------+
|docstring |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|comment   |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|empty     |2      |33.33 |NC       |NC         |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |NC       |NC         |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |NC       |NC         |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |2      |NC       |NC         |
+-----------+-------+---------+-----------+
|refactor   |0      |NC       |NC         |
+-----------+-------+---------+-----------+
|warning    |2      |NC       |NC         |
+-----------+-------+---------+-----------+
|error      |2      |NC       |NC         |
+-----------+-------+---------+-----------+



Messages
--------

+-------------------------+------------+
|message id               |occurrences |
+=========================+============+
|undefined-variable       |2           |
+-------------------------+------------+
|pointless-statement      |2           |
+-------------------------+------------+
|missing-module-docstring |2           |
+-------------------------+------------+




-----------------------------------
Your code has been rated at 0.00/10
```

### Expected behavior
```
************* Module module1
module1.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module1.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module1.py:1:0: E0602: Undefined variable 'foo' (undefined-variable)
************* Module module2
module2.py:1:0: C0114: Missing module docstring (missing-module-docstring)
module2.py:1:0: W0104: Statement seems to have no effect (pointless-statement)
module2.py:1:0: E0602: Undefined variable 'bar' (undefined-variable)


Report
======
2 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |2      |2          |=          |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|method   |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+
|function |0      |NC         |NC         |0           |0        |
+---------+-------+-----------+-----------+------------+---------+



6 lines have been analyzed

Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |4      |66.67 |4        |=          |
+----------+-------+------+---------+-----------+
|docstring |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|comment   |0      |0.00  |NC       |NC         |
+----------+-------+------+---------+-----------+
|empty     |2      |33.33 |2        |=          |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |0          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |2      |2        |2          |
+-----------+-------+---------+-----------+
|refactor   |0      |0        |0          |
+-----------+-------+---------+-----------+
|warning    |2      |2        |2          |
+-----------+-------+---------+-----------+
|error      |2      |2        |2          |
+-----------+-------+---------+-----------+



% errors / warnings by module
-----------------------------

+--------+------+--------+---------+-----------+
|module  |error |warning |refactor |convention |
+========+======+========+=========+===========+
|module2 |50.00 |50.00   |0.00     |50.00      |
+--------+------+--------+---------+-----------+
|module1 |50.00 |50.00   |0.00     |50.00      |
+--------+------+--------+---------+-----------+



Messages
--------

+-------------------------+------------+
|message id               |occurrences |
+=========================+============+
|undefined-variable       |2           |
+-------------------------+------------+
|pointless-statement      |2           |
+-------------------------+------------+
|missing-module-docstring |2           |
+-------------------------+------------+




------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.10.8 (main, Nov 10 2022, 12:59:17) [GCC 11.2.1 20220219]
```

### OS / Environment

Alpine Linux v3.16.3 ([Dockerfile](https://github.com/cytopia/docker-pylint/blob/0.8/Dockerfiles/Dockerfile))
